### DISCUSSION TITLE
Building with openssl

### DISCUSSION THREAD

**Discussion by shaharmor** [uNetworking/uWebSockets.js#804] [Q&A]
Hi,

We're trying to build the uWebSockets.js module with openssl instead of boringssl.
It seems that currently the `make` command only supports building with boringssl. As we're not c/cpp developers, we would greatly appreciate any support in converting the build script to work with openssl as well

**Comment by joshxyzhimself** [comment]
Question: are there any advantages for preferring openssl over boringssl? (I'm curious, there might be technical advantages or business compliance use cases, I'm clueless lol)

**Reply by uNetworkingAB** [reply]
BoringSSL is a lot faster than OpenSSL for small messages while OpenSSL is faster on large ones. BoringSSL can easily be statically compiled and has support for quic (which we need for http3). So openssl support is not really there anymore.

**Reply by shaharmor** [reply]
The main reason we need it is because as far as I know, boringssl doesn’t support changing the ciphers used, and because there are some older ciphers there we can’t use it.

we wanted to use OpenSSL so we can control the ciphers used.

Now that I think of it, if boringssl is indeed bundled with the library, we might be able to edit the boringssl code directly to remove the ciphers we don’t want, right?

**Reply by uNetworkingAB** [reply]
We don't support openssl. We use boringssl.
